src/Autofac/Core/Registration/ServiceRegistrationInfo.cs

```diff
@@ -43,6 +43,7 @@ internal class ServiceRegistrationInfo : IResolvePipelineBuilder
         private volatile bool _isInitialized;
 
         [SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields", Justification = "The _service field is useful in debugging and diagnostics.")]
+        [SuppressMessage("CodeQuality", "IDE0052:Remove unread private members", Justification = "The _service field is useful in debugging and diagnostics.")]
```

Comment:
Perhaps something should actually be _done_ with the field. For example, a `ToString` override or a [`DebuggerDisplay` implementation that uses it](https://docs.microsoft.com/en-us/visualstudio/debugger/using-the-debuggerdisplay-attribute?view=vs-2019).

Reply:
Makes sense; used it in a ToString implementation.